### ISSUE TITLE
false positive on op_ref lint

### ISSUE THREAD

**Issue by dimbleby** [rust-lang/rust#41137]
Looking for items in one set but not another, clippy tells me:

```
warning: taken reference of both operands, which is done automatically by the operator anyway
  --> examples/unix/cares_event_loop.rs:88:24
   |
88 |         let unwanted = &tracked_fds - &new_fds;
   |                        ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(op_ref)] on by default
help: use the values directly
   |         let unwanted = tracked_fds - new_fds;
   = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#op_ref
```

But if I follow that advice, the compiler tells me:

```
error[E0369]: binary operation `-` cannot be applied to type `std::collections::HashSet<i32>`
  --> examples/unix/cares_event_loop.rs:88:24
   |
88 |         let unwanted = tracked_fds - new_fds;
   |                        ^^^^^^^^^^^
   |
note: an implementation of `std::ops::Sub` might be missing for `std::collections::HashSet<i32>`
  --> examples/unix/cares_event_loop.rs:88:24
   |
88 |         let unwanted = tracked_fds - new_fds;
   |                        ^^^^^^^^^^^

error: aborting due to previous error
```

(Also, the "for further information" link for this lint does not exist).

**Comment by dimbleby** [comment]
oh sorry, wrong repo, this is supposed to be in clippy.
